R/get_generic.R

```diff
@@ -0,0 +1,72 @@
+## function name get_generic
+
+#' retrieve all parameters of specified type
+#' @param data parameter dataframe outputted from load_epidata()
+#' @param parameter_name name of the parameter type to retrieve, ensuring the name matches that in data
+#' @return dataframe with all parameter estimates and specified columns
+#' @examples
+#' df <- load_epidata(pathogen = "ebola")
+#' get_generic(data = df$params, parameter_name = "Human delay - serial interval")
+#' @examples
+#' df <- load_epidata(pathogen = "marburg")
+#' get_generic(data = df$params, parameter_name = "Attack rate")
+get_generic <- function(data, parameter_name) {
+  if(length(data[which(data$parameter_type == parameter_name),]) == 0) {
+    stop("data should be the parameter dataframe outputted from load_epidata()")
+  } else {
+    index <- which(data$parameter_type == parameter_name)
+    pathogen <- unique(data$pathogen)
+    if(pathogen == "Ebola virus") {
```

Comment:
I did it this way because columns are ordered in different ways and we also need different columns for different pathogens. I'm a little confused by the suggestion regarding the article column type -- I am not trying to filter by specific types but rather the columns that I thought a user would want to output. Will discuss with others if there is a more sophisticated way that we can do this though that scales to other pathogens.